sdk/core/azure-core/inc/azure/core/http/http.hpp

```diff
@@ -40,6 +40,21 @@ namespace Azure { namespace Core { namespace Http {
       left.insert(right.begin(), right.end());
       return left;
     }
+
+    /**
+     * @brief Insert a header into \p headers checking that \p headerName does not contains valid
```

Comment:
Do we really want to reject valid characters? 😜

```suggestion
     * @brief Insert a header into \p headers checking that \p headerName does not contain invalid
```